Functions/xASL_bids_parms2BIDS.m

```diff
@@ -327,8 +321,53 @@
 	end
 end
 
-% Move fields to correct position in x structure
-outParms = xASL_io_CheckDeprecatedFieldsX(outParms);
+% If we convert to xASL format, we have to move fields to correct position in the x structure
+if ~bOutBids
+    outParms = xASL_io_CheckDeprecatedFieldsX(outParms);
+end
 
 
 end
+
+
+% Flatten the x structure to a single level
+function inXasl = xASL_wrp_Quantify_FlattenX(inXasl)
+
+    % Flatten inXasl structure
+    
+    % Move x.Q, x.settings, x.dataset, x.modules, x.modules.structural,
+    % x.modules.asl, and x.modules.population parameters to main level of x
+    
+    firstLevelStructs = {'Q', 'settings', 'dataset', 'modules'};
+    secondLevelStructs = {'structural', 'asl', 'population'};
+    
+    % Iterate over first level structs
+    for iStruct = 1:numel(firstLevelStructs)
+        if isfield(inXasl,firstLevelStructs{iStruct})
+            FieldsQ = fields(inXasl.(firstLevelStructs{iStruct}));
+            for iField=1:length(FieldsQ)
+                inXasl.(FieldsQ{iField}) = inXasl.(firstLevelStructs{iStruct}).(FieldsQ{iField});
+            end
+            inXasl = rmfield(inXasl,firstLevelStructs{iStruct});
+        end
+    end
+    
+    % Iterate over second level structs
+    if isfield(inXasl,'modules')
+        for iStruct = 1:numel(secondLevelStructs)
+            if isfield(inXasl.modules,secondLevelStructs{iStruct})
+                FieldsQ = fields(inXasl.modules.(secondLevelStructs{iStruct}));
+                for iField=1:length(FieldsQ)
+                    inXasl.(FieldsQ{iField}) = inXasl.modules.(secondLevelStructs{iStruct}).(FieldsQ{iField});
+                end
+                inXasl = rmfield(inXasl.modules,secondLevelStructs{iStruct});
```

Comment:
error!! `inxAasl.modules = rmfield(inXasl.modules,....)`

Or even better - move this one cycle below and delete the whole `modules` stuff. 
`inXasl = rmfield(inXasl,'modules')`

Reply:
Oh damn... I wasn't thinking of that. I'm not even sure if we have to move fields from there to the base level anyway, since it's mostly processing settings so far. Let's talk later in our coding session.

Reply:
147a173